compiler/noirc_evaluator/src/acir/acir_context/mod.rs

```diff
@@ -1166,8 +1166,10 @@ impl<F: AcirField> AcirContext<F> {
         let assert_message =
             self.generate_assertion_message_payload("Attempt to divide with overflow".to_string());
         let unsigned = self.not_var(q_sign, AcirType::unsigned(1))?;
-        // We just use `unsigned` for the predicate of assert_neq_var because if the `predicate` is false, the quotient
-        // we get from the unsigned division under the predicate will not be 2^{bit_size-1} anyways.
+
+        // This overflow check must also be under the predicate
+        let unsigned = self.mul_var(unsigned, predicate)?;
```

Comment:
There was a comment here about why we weren't using the predicate var. However, not using it seemed wrong? And using it fixes things so I don't understand the original comment.

Reply:
This comment looks wrong: https://github.com/noir-lang/noir/blob/31e16d9691ab45206f17ceda0d477ed8d9624d35/compiler/noirc_evaluator/src/acir/acir_context/mod.rs#L1131-L1132 

Shouldn't it have divided by `sign(a)`?

Reply:
I can try it... but wouldn't it be incorrect to not check for unsigned? Otherwise I think if it's a signed value and the underlying field element matches `2^{bit_size-1}` then it will overflow? Not sure...

Reply:
Ah I'm dumb. I was thinking cause we are in `signed_division_var` it was fine, but we do still need to check the sign.